staging/src/k8s.io/apimachinery/pkg/util/managedfields/gvkparser.go

```diff
@@ -58,7 +59,7 @@ func NewGVKParser(models proto.Models, preserveUnknownFields bool) (*GvkParser,
 	parser := GvkParser{
 		gvks: map[schema.GroupVersionKind]string{},
 	}
-	parser.parser = typed.Parser{Schema: *typeSchema}
```

Comment:
is this really trying to copy the schema, or was it just dereferencing because Parser doesn't hold a pointer to Schema? Does storing a non-pointer Schema field which also has a non-pointer sync.Once field mean we're propagating the copylocks error everywhere we pass around parser? Should Parser hold a pointer to Schema?

Reply:
This was just dereferencing Parser because it did not hold a pointer to Schema.

I changed it to a copy because that did not break any existing usages of Parser, and I did not not know if we wanted to do that right now.

I agree that it would be better that Parser held a pointer to Schema, rather than embedded.

Reply:
changing the internal private once to a pointer would be another way to avoid the lock copy, though it would mean you'd need to initialize nil `once` fields

Reply:
yeah, I think we should turn the schema into a pointer and copy the pointer (even if the change is somewhat large, it's a mechanical change).